### PR TITLE
[Feature] [Flink-K8s-V2] Refactor the lifecycle control of Flink session-mode jobs on Kubernetes 

### LINKED ISSUE #2883: [Feature] [Flink-K8s-V2] Refactor the lifecycle control of Flink session-mode jobs on Kubernetes
It is a subtask of #2879 which includes the following content:
- Adapt [KubernetesNativeSessionClient.doSubmit](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L49) to [FlinkOperator.deploySessionJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L36);
- Adapt [KubernetesNativeSessionClient.doCancel](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L259) to  [FlinkOperator.cancelJob，FlinkOperator.stopJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L51) ;
- Adapt [KubernetesNativeSessionClient.doTriggerSavepoint](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L259) to [FlinkOperator.triggerJobSavepoint](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L56);
- Adapt [KubernetesNativeSessionClient.shutdown](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L207) to [FlinkK8sOperator.delete](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L41);
- Remove the remaining useless code；

Ref: [design doc](https://www.craft.me/s/LrTTzYHqSOLYqY)

### PR DESCRIPTION
**Pull request by caicancai**
Refactor the lifecycle control of Flink session-mode jobs on Kubernetes

## What changes were proposed in this pull request

Issue Number: close #2883 

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

•
[KubernetesNativeSessionClient.doSubmit](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L49) 适配到 [FlinkOperator.deploySessionJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L36);

•
[KubernetesNativeSessionClient.doCancel](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L259) 适配到  [FlinkOperator.cancelJob，FlinkOperator.stopJob](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L51) ;

•
[KubernetesNativeSessionClient.doTriggerSavepoint](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L259) 适配到 [FlinkOperator.triggerJobSavepoint](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L56);

•
[KubernetesNativeSessionClient.shutdown](https://github.com/apache/incubator-streampark/blob/dev/streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesNativeSessionClient.scala#L207) 适配到 [FlinkK8sOperator.delete](https://github.com/Al-assad/streampark-flink-kubernetes-v2/blob/master/src/main/scala/org/apache/streampark/flink/kubernetes/operator/FlinkK8sOperator.scala#L41);

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)
 - no

### CODE REVIEW

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala [hunk 1]
```diff
@@ -0,0 +1,141 @@
+package org.apache.streampark.flink.client.impl
```
Comment (wolfboys) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala:17]: Missing Apache license header...

Comment (caicancai) [reply streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala:17]: Thank you for reminding me. I'm sorry I forgot

**Comment by caicancai** [comment]
cc @wolfboys

**Review by wolfboys — APPROVED** [review]
LGTM

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+object KubernetesSessionClientV2 extends KubernetesClientV2Trait with Logger {
+  @throws[Exception]
+  override def doSubmit(
+      submitRequest: SubmitRequest,
+      flinkConfig: Configuration): SubmitResponse = {
+
+    val richMsg: String => String = s"[flink-submit][appId=${submitRequest.id}] " + _
+
+    submitRequest.checkBuildResult()
+    val buildResult = submitRequest.buildResult.asInstanceOf[K8sAppModeBuildResponse]
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala:46]: should be ShadedBuildResponse

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+      submitReq: SubmitRequest,
+      originFlinkConfig: Configuration,
+      buildResult: K8sAppModeBuildResponse): Either[FailureMessage, FlinkSessionJobDef] = {
+
+    val flinkConfObj = originFlinkConfig.clone()
+
+    val namespace = Option(submitReq.k8sSubmitParam.kubernetesNamespace)
+      .getOrElse("default")
+
+    val name = Option(submitReq.k8sSubmitParam.clusterId)
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala:86]: cluster-id is the deployName of FlinkSessionJob CR, not the name of the CR.

File: streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      .filter(str => StringUtils.isNotBlank(str))
+      .getOrElse(return Left("cluster-id should not be empty"))
+
+    val deploymentName = Option(submitReq.developmentMode.name())
```
Comment (Al-assad) [inline streampark-flink/streampark-flink-client/streampark-flink-client-core/src/main/scala/org/apache/streampark/flink/client/impl/KubernetesSessionClientV2.scala:90]: deployName is actually the cluster-id

**Pull request merged**